style(tooltips): Add border to tooltips

glynnis opened the pull request:
https://jira.rax.io/browse/FRMW-551

- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM

## Before
![screen shot 2015-12-15 at 3 39 48 pm](https://cloud.githubusercontent.com/assets/1529366/11824909/e32f12aa-a341-11e5-84b6-1901ce8c6211.png)
![screen shot 2015-12-15 at 3 39 51 pm](https://cloud.githubusercontent.com/assets/1529366/11824908/e32d12de-a341-11e5-97f8-f4149e87418f.png)

## After
![screen shot 2015-12-22 at 10 54 19 am](https://cloud.githubusercontent.com/assets/1529366/11960886/805c7812-a89a-11e5-8873-d95cc194106e.png)
![screen shot 2015-12-22 at 10 54 26 am](https://cloud.githubusercontent.com/assets/1529366/11960887/805d454e-a89a-11e5-8c2f-4e8e89c1d9b5.png)
![screen shot 2015-12-22 at 10 54 29 am](https://cloud.githubusercontent.com/assets/1529366/11960884/80599886-a89a-11e5-95fa-cff224afb2b8.png)
![screen shot 2015-12-22 at 10 54 32 am](https://cloud.githubusercontent.com/assets/1529366/11960888/805f814c-a89a-11e5-8627-88dc23d17697.png)
![screen shot 2015-12-22 at 10 54 35 am](https://cloud.githubusercontent.com/assets/1529366/11960889/80614d60-a89a-11e5-8a62-50a5c8cbb470.png)

### With a lot of text to demonstrate that arrow placement is still good
![screen shot 2015-12-22 at 10 06 31 am](https://cloud.githubusercontent.com/assets/1529366/11960885/8059ad9e-a89a-11e5-888f-c51f675c683a.png)

glynnis commented on src/styles/vars.less:
```diff
@@ -271,6 +271,19 @@
+@tooltipBg: @gray-700;
+@tooltipOpacity: 1;
+@tooltipArrowWidth: 5px;
+@tooltipArrowColor: @tooltipBg;
```
Moved variables from component LESS to `vars.less`

ssalinas24 commented:
LGTM

glynnis commented:
Adding "do not merge" label because @ElementE found a possible implementation that will allow for the border to extend around the arrow/triangle part of the tooltip as well, so I'm going to investigate.

100stacks commented:
@glynnis @ssalinas24, although this PR is not directly related to https://github.com/rackerlabs/encore-ui/issues/1465 it appears that the `tooltip` popup is centered on the container.  Can you take look at `tooltips.less` for this if have some cycles? :bike:  Also, it occurs in Chrome and FireFox.  Thanks!

glynnis commented on src/components/tooltips/tooltips.less:
```diff
@@ -1,25 +1,9 @@
-//** Tooltip arrow width
-@tooltipArrowWidth: 5px;
-//** Tooltip arrow color
-@tooltipArrowColor: @tooltipBg;
```
Moved component variables to `vars.less`.

glynnis commented on src/styles/vars.less:
```diff
@@ -454,3 +470,15 @@
+@tooltipBg: @gray-700;
+@tooltipOpacity: 1;
+@tooltipArrowWidth: 5px;
+@tooltipArrowColor: @tooltipBg;
```
Old component variable names in deprecated section of `vars.less`.

glynnis commented on src/styles/vars.less:
```diff
@@ -340,6 +340,22 @@
+@tooltip-arrow-color: @tooltip-background-color;
+@tooltip-arrow-border-width: 9px;
+@tooltip-arrow-margin-width: 3px;
+@tooltip-border-color: @gray-600;
```
Component variables have been renamed to follow kebab case standard. Added some new variables to accommodate changes in code (`@tooltip-arrow-border-width`, etc.).

100stacks commented on src/components/tooltips/docs/tooltips.html:
```diff
@@ -3,7 +3,7 @@
         <li><span tooltip="Right tooltip" tooltip-placement="right">Hover over me for a right-side tooltip</span></li>
```
@glynnis, one minor suggestion - update to `2 spaces` for indentation vs `4 spaces` with the recent HTML styleguide changes.

glynnis replied:
Ah! Good catch. Updated :+1:

100stacks commented:
LGTM :+1:

ssalinas24 commented on src/styles/vars.less:
```diff
@@ -340,6 +340,22 @@
+@tooltip-z-index: 1070;
+@tooltip-max-width: 200px;
+@tooltip-text-color: @white;
+@tooltip-header-color: @tooltip-text-color;
```
@glynnis I think this should be `@tooltip-header-text-color`, correct?

glynnis replied:
Dah! Good catch. Fixed :+1:

ssalinas24 commented on src/styles/vars.less:
```diff
@@ -340,6 +340,22 @@
+@tooltip-background-color: @gray-700;
+@tooltip-opacity: 1;
+@tooltip-arrow-width: 7px;
+@tooltip-arrow-color: @tooltip-background-color;
```
@glynnis Should we make this `@tooltip-arrow-border-color`?

glynnis replied:
@ssalinas24 If I understand you correctly, I think this makes sense as is. 

There are two main colors for tooltips: (1) the tooltip color (which is the same as the arrow color), and (2) the border color (which is the same as the arrow border color). Those vars get used by both the arrow and the tooltip box itself.

I suppose I could make an additional var, `@tooltip-arrow-border-color`, but I'm not sure that's necessary -- it'll always be the same as `@tooltip-border-color`.

Let me know if I can clarify or if it seems like I misunderstood your comment :smile:

ElementE commented:
Design LGTM!

CITguy commented on src/components/tooltips/tooltips.less:
```diff
@@ -82,31 +63,66 @@
+
+  // These classes don't appear in the demo and have been deprecated.
+
+  &.left {
```
Though the `.left` class isn't used in the demo, it should still be supported.  However, ng-bootstrap only supports `.top`, `.left`, `.bottom`, and `.right`, so I would remove the `.top-left`, `.bottom-left`, `.top-right`, `.bottom-right` definitions. 

I actually made a tweak to this demo in the `ng1.3` branch to better display the capabilities of placement and functionality. I think it would be appropriate to incorporate those changes in this PR. Please take a look at 0e6a8f8 for the necessary changes.

glynnis commented:
OMG sorry it took me so long to get to this :tired_face: 

I've incorporated relevant changes from 0e6a8f8 and added the Left tooltip to the demo:
![screen shot 2016-01-18 at 3 06 42 pm](https://cloud.githubusercontent.com/assets/1529366/12402876/8d1a2804-bdf5-11e5-9067-9ec5bf5685ec.png)

![screen shot 2016-01-18 at 3 10 06 pm](https://cloud.githubusercontent.com/assets/1529366/12402881/9780780c-bdf5-11e5-9ea7-85d3987736ff.png)

glynnis commented:
Also rebased this.

Droogans commented:
@glynnis you have accidentally added an outdated form of page object documentation (API.md).

Please kill your dev server, make sure you're up to date and stuff, and then re-run it. It shouldn't be building that file anymore!

Once you're certain that's good, please delete that file from the PR and force push it back up.

CITguy commented:
LGTM
